api/test-context/src/test_context.rs

```diff
@@ -438,23 +549,36 @@ impl TestContext {
             .unwrap();
     }
 
+    pub async fn get_apt_balance(&self, account: AccountAddress) -> u64 {
+        let coin_balance = self
+            .api_get_account_resource(
+                account,
+                "0x1",
+                "coin",
+                "CoinStore<0x1::aptos_coin::AptosCoin>",
+            )
+            .await;
+        coin_balance["data"]["coin"]["value"]
+            .as_str()
+            .unwrap()
+            .parse::<u64>()
+            .unwrap()
+    }
+
     // TODO: Add support for generic_type_params if necessary.
     pub async fn api_get_account_resource(
         &self,
-        account: &LocalAccount,
+        account: AccountAddress,
         resource_account_address: &str,
         module: &str,
         name: &str,
     ) -> serde_json::Value {
         let resources = self
-            .get(&format!(
-                "/accounts/{}/resources",
-                account.address().to_hex_literal()
-            ))
+            .get(&format!("/accounts/{}/resources", account.to_hex_literal()))
             .await;
         let vals: Vec<serde_json::Value> = serde_json::from_value(resources).unwrap();
         vals.into_iter()
-            .find(|v| v["type"] == format!("{}::{}::{}", resource_account_address, module, name,))
+            .find(|v| v["type"] == format!("{}::{}::{}", resource_account_address, module, name))
             .unwrap()
     }
```

Comment:
per my own experience, this is a bit of a mess, because there are two ways to get this value... maybe we should have this function query both end points and ensure it is available to both....

Reply:
I updated test_context to refresh the root account's sequence number every time in root_account(). Not the cleanest solution as that makes that function async + requires await everywhere, but it works for now. We can go back and refactor test_context + api tests later.

Reply:
I can leave a TODO here to clean this up in the future